packages/devtools_app/lib/src/shared/memory/class_name.dart

```diff
@@ -72,26 +72,40 @@ enum ClassType {
 }
 
 class HeapClassName {
-  HeapClassName({required this.className, required String? library})
+  HeapClassName._({required String? library, required this.className})
       : library = _normalizeLibrary(library);
 
-  HeapClassName.fromClassRef(ClassRef? classRef)
-      : this(
-          library: _library(
-            classRef?.library?.name,
-            classRef?.library?.uri,
-          ),
-          className: classRef?.name ?? '',
-        );
-
-  HeapClassName.fromHeapSnapshotClass(HeapSnapshotClass? theClass)
-      : this(
-          library: _library(
-            theClass?.libraryName,
-            theClass?.libraryUri.toString(),
-          ),
-          className: theClass?.name ?? '',
-        );
+  static final _instances = <HeapClassName>{};
```

Comment:
Does this set ever need to be cleared?

Reply:
Good point. It should be cleared on reconnect. I can initiate the clearing in MemoryController. What is the best way to catch reconnect event?

Reply:
If you can verify that `MemoryController.dispose()` is called on disconnect, that would be a good place to put it.

Reply:
And if our controllers are not being disposed on disconnect, then that is a bug

Reply:
Looks like they are not being disposed - this is a leak https://github.com/flutter/devtools/issues/6301. 

For this PR, I would make sure that this set of instances is cleared out by proxy of calling `MemoryController.dispose` and then the leak in the issue I referenced should be fixed, which would then clear this set as expected.

Reply:
Thanks. Done.